### PR TITLE
.NET SDK Refresh for Elastic

### PR DESCRIPTION
**Pull request by jkore1911**
.NET SDK Refresh for Elastic

### CODE REVIEW

**Comment by jkore1911** [comment]
@microsoft-github-policy-service agree company="Microsoft"

File: sdk/elastic/Azure.ResourceManager.Elastic/src/autorest.md [hunk 1]
```diff
@@ -53,4 +53,10 @@ acronym-mapping:
 rename-mapping:
   MonitoredResource: MonitoredResourceContent
 
+directive:
+  - from: elastic.json
+    where: $.definitions
+    transform: >
+      $.ElasticTrafficFilter.properties.type['x-ms-client-name']= 'ElasticTrafficFilterType'
```
Comment (ArcturusZhang) [inline sdk/elastic/Azure.ResourceManager.Elastic/src/autorest.md:60]: we usually do not use directive to rename things, could you remove this and use our `rename-mapping` instead?
This should be something like:
```
rename-mapping:
	ElasticTrafficFilter.type: ElasticTrafficFilterType
```

File: sdk/elastic/Azure.ResourceManager.Elastic/api/Azure.ResourceManager.Elastic.netstandard2.0.cs [hunk 2]
```diff
@@ -77,6 +119,52 @@ public ElasticMonitorResourceData(Azure.Core.AzureLocation location) { }
...
+        System.Collections.Generic.IAsyncEnumerator<Azure.ResourceManager.Elastic.MonitoredSubscriptionPropertyResource> System.Collections.Generic.IAsyncEnumerable<Azure.ResourceManager.Elastic.MonitoredSubscriptionPropertyResource>.GetAsyncEnumerator(System.Threading.CancellationToken cancellationToken) { throw null; }
+        System.Collections.Generic.IEnumerator<Azure.ResourceManager.Elastic.MonitoredSubscriptionPropertyResource> System.Collections.Generic.IEnumerable<Azure.ResourceManager.Elastic.MonitoredSubscriptionPropertyResource>.GetEnumerator() { throw null; }
+        System.Collections.IEnumerator System.Collections.IEnumerable.GetEnumerator() { throw null; }
+    }
+    public partial class MonitoredSubscriptionPropertyData : Azure.ResourceManager.Models.ResourceData, System.ClientModel.Primitives.IJsonModel<Azure.ResourceManager.Elastic.MonitoredSubscriptionPropertyData>, System.ClientModel.Primitives.IPersistableModel<Azure.ResourceManager.Elastic.MonitoredSubscriptionPropertyData>
```
Comment (ArcturusZhang) [inline sdk/elastic/Azure.ResourceManager.Elastic/api/Azure.ResourceManager.Elastic.netstandard2.0.cs:139]: `MonitoredSubscriptionProperty` does not make much sense as a resource name.
Shall we rename this to something like `ElasticMoniteredSubscription`?

File: sdk/elastic/Azure.ResourceManager.Elastic/api/Azure.ResourceManager.Elastic.netstandard2.0.cs [hunk 3]
```diff
@@ -142,24 +280,62 @@ protected MockableElasticResourceGroupResource() { }
...
+        public static Azure.ResourceManager.Elastic.Models.UpgradableVersionsList UpgradableVersionsList(string currentVersion = null, System.Collections.Generic.IEnumerable<string> upgradableVersions = null) { throw null; }
+        public static Azure.ResourceManager.Elastic.Models.UserApiKeyResponse UserApiKeyResponse(string apiKey = null) { throw null; }
         public static Azure.ResourceManager.Elastic.Models.VmIngestionDetailsResponse VmIngestionDetailsResponse(string cloudId = null, string ingestionKey = null) { throw null; }
         public static Azure.ResourceManager.Elastic.Models.VmResources VmResources(string vmResourceId = null) { throw null; }
     }
+    public partial class BillingInfoResponse : System.ClientModel.Primitives.IJsonModel<Azure.ResourceManager.Elastic.Models.BillingInfoResponse>, System.ClientModel.Primitives.IPersistableModel<Azure.ResourceManager.Elastic.Models.BillingInfoResponse>
```
Comment (ArcturusZhang) [inline sdk/elastic/Azure.ResourceManager.Elastic/api/Azure.ResourceManager.Elastic.netstandard2.0.cs:328]: We should not have a type ending with `Response`.
Could we rename this to something like `ElasticBillingInfoResult`?

**Comment by cijothomas** [comment]
[jkore1911](https://github.com/jkore1911) requested review from [cijothomas](https://github.com/cijothomas), [reyang](https://github.com/reyang), [rajkumar-rangaraj](https://github.com/rajkumar-rangaraj), [TimothyMothra](https://github.com/TimothyMothra), [xiang17](https://github.com/xiang17), [mconnew](https://github.com/mconnew) and [subhra231](https://github.com/subhra231) as [code owners](https://github.com/Azure/azure-sdk-for-net/blob/2e4012bcf5fe519ed484a41bcf8ddfcda5d278ce/.github/CODEOWNERS#L666) [1 hour ago](https://github.com/Azure/azure-sdk-for-net/pull/45930#event-14299993061)

Not sure why? I have nothing to do with Elastic!

File: sdk/elastic/Azure.ResourceManager.Elastic/api/Azure.ResourceManager.Elastic.netstandard2.0.cs (hunk 2 above)
Comment (jkore1911) [reply sdk/elastic/Azure.ResourceManager.Elastic/api/Azure.ResourceManager.Elastic.netstandard2.0.cs:139]: For now, let's go with "MonitoredSubscriptionProperty" as the resource name.

**Comment by ArcturusZhang** [comment]
> [jkore1911](https://github.com/jkore1911) requested review from [cijothomas](https://github.com/cijothomas), [reyang](https://github.com/reyang), [rajkumar-rangaraj](https://github.com/rajkumar-rangaraj), [TimothyMothra](https://github.com/TimothyMothra), [xiang17](https://github.com/xiang17), [mconnew](https://github.com/mconnew) and [subhra231](https://github.com/subhra231) as [code owners](https://github.com/Azure/azure-sdk-for-net/blob/2e4012bcf5fe519ed484a41bcf8ddfcda5d278ce/.github/CODEOWNERS#L666) [1 hour ago](https://github.com/Azure/azure-sdk-for-net/pull/45930#event-14299993061)
> 
> Not sure why? I have nothing to do with Elastic!

Maybe because the extra api files.
@jkore1911 could you remove those files?

File: sdk/elastic/Azure.ResourceManager.Elastic/src/autorest.md (hunk 1 above)
Comment (jkore1911) [reply sdk/elastic/Azure.ResourceManager.Elastic/src/autorest.md]: Done

File: sdk/elastic/Azure.ResourceManager.Elastic/api/Azure.ResourceManager.Elastic.netstandard2.0.cs (hunk 3 above)
Comment (ArthurMa1978) [inline sdk/elastic/Azure.ResourceManager.Elastic/api/Azure.ResourceManager.Elastic.netstandard2.0.cs:328]: Consider rename to BillingInfoResult

File: sdk/elastic/Azure.ResourceManager.Elastic/api/Azure.ResourceManager.Elastic.netstandard2.0.cs [hunk 4]
```diff
@@ -415,6 +856,38 @@ public MonitorProperties() { }
...
+        [System.ComponentModel.EditorBrowsableAttribute(System.ComponentModel.EditorBrowsableState.Never)]
+        public override bool Equals(object obj) { throw null; }
+        [System.ComponentModel.EditorBrowsableAttribute(System.ComponentModel.EditorBrowsableState.Never)]
+        public override int GetHashCode() { throw null; }
+        public static bool operator ==(Azure.ResourceManager.Elastic.Models.Status left, Azure.ResourceManager.Elastic.Models.Status right) { throw null; }
+        public static implicit operator Azure.ResourceManager.Elastic.Models.Status (string value) { throw null; }
+        public static bool operator !=(Azure.ResourceManager.Elastic.Models.Status left, Azure.ResourceManager.Elastic.Models.Status right) { throw null; }
+        public override string ToString() { throw null; }
+    }
+    public partial class SubscriptionList : System.ClientModel.Primitives.IJsonModel<Azure.ResourceManager.Elastic.Models.SubscriptionList>, System.ClientModel.Primitives.IPersistableModel<Azure.ResourceManager.Elastic.Models.SubscriptionList>
```
Comment (ArthurMa1978) [inline sdk/elastic/Azure.ResourceManager.Elastic/api/Azure.ResourceManager.Elastic.netstandard2.0.cs:878]: The name is too generic, please provide a more descriptive name.

**Review by ArthurMa1978 — APPROVED** [review]

**Review by ArthurMa1978 — APPROVED** [review]

**Pull request merged**